src/js/containers/aboutTheData/AgencyDetailsContainer.jsx

```diff
@@ -282,38 +123,89 @@ const AgencyDetailsContainer = ({ modalClick, agencyName }) => {
     const verticalStickyClass = isVertialSticky ? 'sticky-y-table' : '';
     const horizontalStickyClass = isHorizontalSticky ? 'sticky-x-table' : '';
 
-    const rows = mockAPIResponse.results.map(
-        ({
-            fiscal_year: fiscalYear,
-            percent_total_budget: total,
-            recent_update: recentUpdate,
-            discrepancy_count: missingTasCount,
-            obligation_difference: obligationDiff,
-            unlinked_cont_awd: unlinkedCont,
-            unlinked_asst_awd: unlinkedAsst,
-            assurance_statement_file: assuranceStatementFile,
-            tas_account_discrepancies_totals: tasTotals
-        }) => [
-            (<div className="generic-cell-content">{ fiscalYear }</div>),
-            (<div className="generic-cell-content">{ total }</div>),
-            (<CellWithModal data={recentUpdate} openModal={modalClick} modalType="publicationDates" agencyData={{ agencyName }} />),
-            (<CellWithModal data={missingTasCount} openModal={modalClick} modalType="missingAccountBalance" agencyData={{ agencyName, gtasObligationTotal: tasTotals.gtas_obligation_total }} />),
-            (<CellWithModal data={obligationDiff} openModal={modalClick} modalType="reportingDifferences" agencyData={{ agencyName }} />),
-            (<div className="generic-cell-content">{ unlinkedCont }</div>),
-            (<div className="generic-cell-content">{ unlinkedAsst }</div>),
-            (<div className="generic-cell-content"><AgencyDownloadLinkCell file={assuranceStatementFile} /></div>)
-        ]
+    const parseRows = (results, federalBudget) => (
+        results.map((row) => {
+            const rowData = Object.create(BaseReportingPeriodRow);
+            rowData.populate(row, federalBudget);
+            return ([
+                (<div className="generic-cell-content">{ rowData.reportingPeriod }</div>),
+                (<div className="generic-cell-content">{ rowData.percentOfBudget }</div>),
+                (<CellWithModal data={rowData.mostRecentUpdate} openModal={modalClick} modalType="publicationDates" agencyName={agencyName} />),
+                (<CellWithModal data={rowData.missingTAS} openModal={modalClick} modalType="missingAccountBalance" agencyName={agencyName} />),
+                (<div className="generic-cell-content">{ rowData.obligationDifference }</div>),
+                (<div className="generic-cell-content">--</div>),
+                (<div className="generic-cell-content">--</div>),
```

Comment:
These columns were assumed to be post-MVP when API contracts were written, and will be added in a later PR when data is available.